### PR TITLE
Move hero image preload out of experimental

### PR DESCRIPTION
**Pull request by sebastianbenz**
This also changes the transformer to only run if there a no pre-existing image preloads.

### CODE REVIEW

File: packages/optimizer/lib/transformers/PreloadHeroImage.js [hunk 1]
```diff
@@ -78,6 +83,28 @@ class PreloadHeroImage {
...
   }
 
+  hasExistingImagePreload(head) {
+    for (const node of head.children) {
+      if (this.isImagePreload(node)) {
+        return true;
+      }
+    }
+    return false;
+  }
```
Comment (patrickkettner) [inline packages/optimizer/lib/transformers/PreloadHeroImage.js:93]: continuing my series of "unhelpful comments that make the code less readable"...

```js
return [...head.children].any(this.isImagePreload)
```

File: packages/optimizer/spec/end-to-end/body-only/expected_output.default.html [hunk 2]
```diff
@@ -6,6 +6,7 @@
   <link rel="preload" href="https://cdn.ampproject.org/v0.js" as="script">
   <script data-auto async src="https://cdn.ampproject.org/v0.js"></script>
   <script async src="https://cdn.ampproject.org/v0/amp-video-0.1.js" custom-element="amp-video"></script>
+  <link rel="preload" href="https://amp.dev/static/samples/img/tokyo.jpg" as="image" data-hero>
```
Comment (patrickkettner) [inline packages/optimizer/spec/end-to-end/body-only/expected_output.default.html:9]: just for my own edification - what is the purpose of the data attr?

File: packages/optimizer/lib/transformers/PreloadHeroImage.js (hunk 1 above)
Comment (sebastianbenz) [reply packages/optimizer/lib/transformers/PreloadHeroImage.js:93]: Why the destructuring? I think `head.children.some(this.isImagePreload);` does the trick as well (and is easy to read).

File: packages/optimizer/spec/end-to-end/body-only/expected_output.default.html (hunk 2 above)
Comment (sebastianbenz) [reply packages/optimizer/spec/end-to-end/body-only/expected_output.default.html:9]: For being able to track which images have been identified as hero images

**Comment by westonruter** [comment]
For images that have `srcset` and `sizes`, this may not work as expected: #823.

**Pull request merged**